src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.AsyncMethodToClassRewriter.Spilling.vb

```diff
@@ -298,8 +301,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
                         array = array.Update(spilledExpression, spilledIndices.AsImmutableOrNull, array.IsLValue, array.Type)
 
-                        ' Make sure side effects are checked
-                        builder.AddStatement(Me.F.ExpressionStatement(array))
+                        If evaluateSideEffects Then
```

Comment:
>If evaluateSideEffects Then [](start = 24, length = 27)

Do we want to evaluate side-effects before evaluating the RHS when we assign directly to the array element? This looks inconsistent with what C# is doing.